File: backoffice/src/product/domain/exceptions/InvalidProductNameException.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain import DomainException

"""
 *
 * Class 
 *
"""

class InvalidProductNameException( DomainException ):

    """
     *
     * Constants 
     *
    """

    __INVALID_PRODUCT_NAME = 124

    """
     *
     * Methods 
     *
    """

    def __init__( self, name : str ) -> None:
        super().__init__( 
            self.__INVALID_PRODUCT_NAME,
            'The product name {} is invalid'.format( name )
        )

File: preparation/src/product/domain/exceptions/InvalidProductNameException.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.product.domain {

    public class InvalidProductNameException : DomainException {

        /* 
         * 
         * Constants
         *
        */

        private const int INVALID_PRODUCT_NAME = 124;

        /* 
         * 
         * Methods
         *
        */

        public InvalidProductNameException( string name ) : base( 
            INVALID_PRODUCT_NAME,
            string.Format( "The product name {0} is invalid", name )
        ) {
        }

    }

}